pkg/codegen/python/gen.go

```diff
@@ -1275,6 +1291,152 @@ func (mod *modContext) genProperties(w io.Writer, properties []*schema.Property,
 	}
 }
 
+func (mod *modContext) genMethods(w io.Writer, res *schema.Resource) {
```

Comment:
I was originally going to modify `genFunction` to also support emitting methods, but decided to have a separate implementation for methods to avoid unnecessary diffs with #7087. After that lands, we may be able to do some consolidation.